src/main/java/com/google/sps/servlets/NewPostServlet.java

```diff
@@ -0,0 +1,64 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.Post;
+import com.google.sps.data.Post.Gender;
+import com.google.sps.data.Post.Status;
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+
+@WebServlet("/new-post") // open to changes in what it's called! 
+public class NewPostServlet extends HttpServlet {
+    
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // would sanitisation be necessary here?
+        // the names inside the getParameter should match with html element attributes 
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+        String petName = request.getParameter("pet_name");
+        String location = request.getParameter("location");
+        String animalType = request.getParameter("animal_type");
+        String breed = request.getParameter("breed");
+        String dob = request.getParameter("birthday"); 
```

Comment:
Thank you! I frankly was a little confused by RFC 3339 so I used ResolverStyle.STRICT from LocalDate parsing instead—please let me know though if you think doing something more related to RFC might be better! (refer to commit [e2af91a])

On a different note, I noticed I'm getting errors in "package com.google.sps.servlets;". This doesn't seem related to the new changes I made in this commit because when I delete the lines 41-42 the error doesn't go away. 
The error says 
`Cannot read the source from /portfolio/src/main/java/com/google/sps/servlets/NewPostServlet.java due to internal exception java.io.IOException:Buffer is closedJava(536871614)`
and I've tried searching Google, but I haven't had successes with solutions to counter this so far. Do you know what might be wrong?